### PR TITLE
Added broadcast variables support to the optimizer.

### PR DESCRIPTION
**Pull request by aalexandrov**
This is the third an final PR for issue #66 after PR #383 and PR #418. With this PR broadcast variables should be supported in trivial dataflows where the broadcast input is within the same iteration scope.

Please check [my comment](https://github.com/stratosphere/stratosphere/pull/383#issuecomment-33047450) in PR #383 for a proposed solution in order to add proper support for broadcast variables with nested iterations.

### CODE REVIEW

File: stratosphere-tests/src/test/java/eu/stratosphere/test/broadcastvars/BoradcastVarsNepheleITCase.java [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+import eu.stratosphere.pact.runtime.task.RegularPactTask;
+import eu.stratosphere.pact.runtime.task.util.LocalStrategy;
+import eu.stratosphere.pact.runtime.task.util.TaskConfig;
+import eu.stratosphere.test.iterative.nephele.JobGraphUtils;
+import eu.stratosphere.test.util.TestBase2;
+import eu.stratosphere.types.LongValue;
+import eu.stratosphere.types.Record;
+import eu.stratosphere.util.Collector;
+
+public class BoradcastVarsNepheleITCase extends TestBase2 {
```
Comment (fhueske) [inline stratosphere-tests/src/test/java/eu/stratosphere/test/broadcastvars/BoradcastVarsNepheleITCase.java]: There is typo in class name. Should be BoraTcastVarsNepheleITCase, no? :-D

Comment (aalexandrov) [inline stratosphere-tests/src/test/java/eu/stratosphere/test/broadcastvars/BoradcastVarsNepheleITCase.java]: D'OH!

**Comment by rmetzger** [comment]
I closed the other broadcast var. pull requests.
You can add code to pending pull requests by pushing into the branch the PR is based on! Travis will test your changes on each push.
I suggest to keep this PR for the broadcast vars until they are finished.

**Comment by aalexandrov** [comment]
I delibarately separated it in three PRs because I wanted to have separate discussions for each part of the implementation (API, runtime, optimizer). Of course, you only have to merge the last one.

**Comment by aalexandrov** [comment]
Will Travis check for the latest commit in this PR 3d53e30?

**Comment by rmetzger** [comment]
It should build on each push (also forced push)

**Comment by StephanEwen** [comment]
Merged in 196c2f1dbabcf3593a25cb8c60e6ff4c6e0da502

Thanks for the Pull Request!

**Pull request closed without merging**